src/main/java/seedu/entities/User.java

```diff
@@ -0,0 +1,153 @@
+        this.targetWeight = 0;
+    }
+
+    public User(String name, float weight, float height, int age, String gender, float targetWeight) {
+        this.name = name;
+        this.weight = weight;
+        this.height = height;
+        this.age = age;
+        this.gender = gender;
+        this.targetWeight = targetWeight;
+        this.caloricLimit = calculateCaloricNeeds(weight, height, age, gender);
+    }
+
+    public String getName() {
+        return this.name;
+    }
+
+    public void setCaloricLimit(double caloricLimit) {
+        this.caloricLimit = caloricLimit;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public float getWeight() {
+        return this.weight;
+    }
+
+    public void setWeight(float weight) {
+        this.weight = weight;
+    }
+    public float getTargetWeight() {
+        return this.targetWeight;
+    }
+    public void setTargetWeight(float targetWeight) {
+        this.targetWeight = targetWeight;
+    }
+
+    public float getHeight() {
+        return this.height;
+    }
+
+    public void setHeight(float height) {
+        this.height = height;
+    }
+
+    public int getAge() {
+        return this.age;
+    }
+
+    public void setAge(int age) {
+        this.age = age;
+    }
+
+    public String getGender() {
+        return this.gender;
+    }
+
+    public void setGender(String gender) {
+        this.gender = gender;
+    }
+
+    public static void displayNewWeightDifference (float weight, float targetWeight) {
+        float actualDifference = weight - targetWeight;
+        float difference = 0;
+        if (actualDifference > 0) {
+            difference = actualDifference;
+            System.out.println("You need to lose " + difference + "kg to reach your target weight: " +
+                    targetWeight + "kg");
+        } else if (actualDifference < 0) {
+            difference = actualDifference * (-1);
+            System.out.println("You need to gain " + difference + "kg to reach your target weight: " +
+                    targetWeight + "kg");
+        } else {
+            System.out.println("Congrats! You have reached your target weight!");
+        }
+    }
+
+    public static void displayNewTargetWeightDifference (float weight, float targetWeight) {
+        float actualDifference = weight - targetWeight;
+        float difference = 0;
+        if (actualDifference > 0) {
+            difference = actualDifference;
+            System.out.println("With your updated target weight, you now need to lose " + difference +
+                    "kg to reach it.");
+        } else if (actualDifference < 0) {
+            difference = actualDifference * (-1);
+            System.out.println("With your updated target weight, you now need to gain " + difference +
+                    "kg to reach it.");
+        } else {
+            System.out.println("Congrats! You have reached your target weight!");
+        }
+    }
+
```

Comment:
The above two function seem to be the same other than the output statement. Consider combine the function with parameter to differentiate the output